diff --git a/src/Util/Getopt.php b/src/Util/Getopt.php
index ba21be3..230849c 100644
--- a/src/Util/Getopt.php
+++ b/src/Util/Getopt.php
@@ -35,7 +35,7 @@ class PHPUnit_Util_Getopt
         reset($args);
         array_map('trim', $args);
 
-        while (list($i, $arg) = each($args)) {
+        foreach ($args as $i => $arg) {
             if ($arg == '') {
                 continue;
             }
@@ -94,7 +94,7 @@ class PHPUnit_Util_Getopt
                     if ($i + 1 < $argLen) {
                         $opts[] = array($opt, substr($arg, $i + 1));
                         break;
-                    } elseif (list(, $opt_arg) = each($args)) {
+                    } elseif ($opt_arg = current($args)) {
                     } else {
                         throw new PHPUnit_Framework_Exception(
                             "option requires an argument -- $opt"
@@ -140,7 +140,7 @@ class PHPUnit_Util_Getopt
             if (substr($long_opt, -1) == '=') {
                 if (substr($long_opt, -2) != '==') {
                     if (!strlen($opt_arg) &&
-                        !(list(, $opt_arg) = each($args))) {
+                        !($opt_arg = current($args))) {
                         throw new PHPUnit_Framework_Exception(
                             "option --$opt requires an argument"
                         );
